warehouse/query-core/src/main/java/datawave/query/jexl/visitors/QueryOptionsFromQueryVisitor.java

```diff
@@ -121,6 +124,65 @@ private Object visitJunction(JexlNode node, Object data, Supplier<JexlNode> crea
         }
     }
     
+    public enum UniqueFunction {
+        UNIQUE_BY_DAY(UniqueFunction.UNIQUE_BY_DAY_FUNCTION, UniqueGranularity.TRUNCATE_TEMPORAL_TO_DAY), UNIQUE_BY_HOUR(
+                        UniqueFunction.UNIQUE_BY_HOUR_FUNCTION, UniqueGranularity.TRUNCATE_TEMPORAL_TO_HOUR), UNIQUE_BY_MILLISECOND(
+                        UniqueFunction.UNIQUE_BY_MILLISECOND_FUNCTION, UniqueGranularity.TRUNCATE_TEMPORAL_TO_MILLISECOND), UNIQUE_BY_MINUTE(
+                        UniqueFunction.UNIQUE_BY_MINUTE_FUNCTION, UniqueGranularity.TRUNCATE_TEMPORAL_TO_MINUTE), UNIQUE_BY_MONTH(
+                        UniqueFunction.UNIQUE_BY_MONTH_FUNCTION, UniqueGranularity.TRUNCATE_TEMPORAL_TO_MONTH), UNIQUE_BY_SECOND(
+                        UniqueFunction.UNIQUE_BY_SECOND_FUNCTION, UniqueGranularity.TRUNCATE_TEMPORAL_TO_SECOND), UNIQUE_BY_TENTH_OF_HOUR(
+                        UniqueFunction.UNIQUE_BY_TENTH_OF_HOUR_FUNCTION, UniqueGranularity.TRUNCATE_TEMPORAL_TO_TENTH_OF_HOUR), UNIQUE_BY_YEAR(
+                        UniqueFunction.UNIQUE_BY_YEAR_FUNCTION, UniqueGranularity.TRUNCATE_TEMPORAL_TO_YEAR);
+        
+        public String getName() {
+            return name;
+        }
+        
+        public final String name;
+        public final UniqueGranularity function;
+        public static final String UNIQUE_BY_DAY_FUNCTION = "unique_by_day";
+        public static final String UNIQUE_BY_HOUR_FUNCTION = "unique_by_hour";
+        public static final String UNIQUE_BY_MINUTE_FUNCTION = "unique_by_minute";
+        public static final String UNIQUE_BY_TENTH_OF_HOUR_FUNCTION = "unique_by_tenth_of_hour";
+        public static final String UNIQUE_BY_MONTH_FUNCTION = "unique_by_month";
+        public static final String UNIQUE_BY_SECOND_FUNCTION = "unique_by_second";
+        public static final String UNIQUE_BY_MILLISECOND_FUNCTION = "unique_by_millisecond";
+        public static final String UNIQUE_BY_YEAR_FUNCTION = "unique_by_year";
+        
+        UniqueFunction(String name, UniqueGranularity function) {
+            this.name = name;
+            this.function = function;
+        }
+        
+        public static UniqueFunction findByName(String name) {
+            switch (name) {
```

Comment:
Not a huge deal, but this could simply be:

```
  return UniqueFunction.valueOf(name.toUpperCase());
```

Reply:
Cool, fixed